services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/volumes.py

```diff
@@ -75,6 +76,23 @@ def _get_s3_volume_driver_config(
     return driver_config
 
 
+def _get_efs_volume_driver_config(
+    efs_settings: AwsEfsSettings,
+    project_id: ProjectID,
+    node_uuid: NodeID,
+    storage_directory_name: str,
+) -> dict[str, Any]:
+    assert "/" not in storage_directory_name  # nosec
+    driver_config: dict[str, Any] = {
+        "Options": {
+            "type": "nfs",
+            "o": f"addr={efs_settings.EFS_DNS_NAME},rw,nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=600,retrans=2,noresvport",
```

Comment:
I se hard coded constants here,all separated by comas.

I would advise to move all these options to the settings and have for each of them a description.
This will allow us to teak without making any changes.

Reply:
parametrized :heavy_check_mark:

Reply:
That would be considered a best practice and I would have added this comment here if ANdrei didnt already :--)

In my mind: If your feature is **clearly** marked a DEV-Frature (right now it is isnt), I dont care about hardcoded strings, they are so to say your own DEV-problem ;) 

But for proper features, that might propagate to prod, I would really prefer to have all "magic numbers" as settings/env vars. In case of issues we can very quickly adjust using portainer in this case, for hardcoded things we are out of luck. To me from where I am sitting, moving all hardcoded numbers (within reason) to be env-vars gives a lot of speed and agility in critical incident/debugging situations at very low overhead for the backend-developers